internal/proxy/channels_mgr.go

```diff
@@ -234,35 +228,38 @@ func decPChanMetrics(pchans []pChan) {
 }
 
 // createMsgStream create message stream for specified collection. Idempotent.
-// If stream already exists, directly return nil.
-func (mgr *singleTypeChannelsMgr) createMsgStream(collectionID UniqueID) error {
-	if mgr.streamExist(collectionID) {
-		log.Info("stream already exist, no need to re-create", zap.Int64("collection_id", collectionID))
-		return nil
+// If stream already exists, directly return it and no error will be returned.
+func (mgr *singleTypeChannelsMgr) createMsgStream(collectionID UniqueID) (msgstream.MsgStream, error) {
+	mgr.mu.RLock()
+	infos, ok := mgr.infos[collectionID]
+	if ok && infos.stream != nil {
+		// already exist.
+		mgr.mu.RUnlock()
+		return infos.stream, nil
 	}
+	mgr.mu.RUnlock()
 
 	channelInfos, err := mgr.getChannelsFunc(collectionID)
 	if err != nil {
+		// What if stream created by other goroutines?
 		log.Error("failed to get channels", zap.Error(err), zap.Int64("collection", collectionID))
-		return err
+		return nil, err
 	}
 
 	stream, err := createStream(mgr.msgStreamFactory, mgr.singleStreamType, channelInfos.pchans, mgr.repackFunc)
 	if err != nil {
+		// What if stream created by other goroutines?
 		log.Error("failed to create message stream", zap.Error(err), zap.Int64("collection", collectionID))
-		return err
+		return nil, err
 	}
 
 	mgr.updateCollection(collectionID, channelInfos, stream)
```

Comment:
This function may be failed, because another coroutine have preemptively set it up.

you can move updateCollection's logic  to this line and delete the updateCollection function.

Reply:
fixed.